File: Water/H5Api/src/main/OpenDataApi/infrastructure/UserToken.java
package OpenDataApi.infrastructure;

import OpenDataAccess.Utility.Common;
import OpenDataAccess.Utility.EnDecrypt;
import OpenDataAccess.Utility.JsonParse;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

public class UserToken {
    private static final String _DesKey = "userlogintokedeskey12345";
    private static final long _Exprie = 86400000;

    public static String CreateToken(String userId) {
        Map<String, Object> map = new HashMap<>();
        map.put("UserId", userId);
        map.put("LoginDate", new Date());
        try {
            return EnDecrypt.DesEncrypt(JsonParse.ToJson(map), _DesKey);
        } catch (Exception ex) {
            return "";
        }
    }

    public static String ParseToken(String token) {
        if (Common.IsNullOrEmpty(token)) return "";
        String content = EnDecrypt.DesDecrypt(token, _DesKey);
        try {
            Map<String, Object> map = JsonParse.JsonToDictionary(content);
            Date loginDate = (Date) Common.ChangeType(Date.class, map.get("LoginDate"));
            if (loginDate.getTime() + _Exprie > new Date().getTime()) return (String) map.get("UserId");
            return "";
        } catch (Exception ex) {
            return "";
        }
    }
}

File: WorkReport/OpenDataAccess/OpenDataAccess.Service/UserToken.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using OpenDataAccess.Utility;

namespace OpenDataAccess.Service
{
    public class UserToken
    {
        private const String _DesKey = "userlogintokedeskey12345";

        static int _Exprie { get; set; }

        static UserToken()
        {
            _Exprie = Common.GetAppSettingsValue<int>("TokenExprie", 1);
        }

        public static string CreateToken(string userId)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict.Add("UserId", userId);
            dict.Add("LoginDate", DateTime.Now);

            return DESEncryptor.Encrypt(Common.ToJson(dict), _DesKey);
        }

        public static string ParseToken(string token)
        {
            if (string.IsNullOrEmpty(token)) return string.Empty;
            try
            {
                string content = DESEncryptor.Decrypt(token, _DesKey);

                Dictionary<string, object> dict = Common.JsonToDictionary(content);
                DateTime loginDate = dict.GetValue<DateTime>("LoginDate");

                if (loginDate.AddDays(_Exprie) > DateTime.Now) return dict.GetStringValue("UserId");
            }
            catch
            {
                throw new Exception("Token解析异常！");
            }

            throw new Exception("Token过期！");
        }
    }
}
